ODLA/platforms/odla_popart/odla_compute.cc

```diff
@@ -147,28 +167,37 @@ odla_status odla_CreateContext(odla_context* context) {
 
   // Create InferenceSession
   auto proto = g_comp->builder->getModelProto();
-  auto session = popart::InferenceSession::createFromOnnxModel(
+  g_comp->session = popart::InferenceSession::createFromOnnxModel(
```

Comment:
can we check if g_comp->session has already been created or not?
we will call 
 "odla_CreatContext() -- odla_DestroyContext() " multiple times.

The "inferenceSession" has the same lifecycle as computation, is that correct?
we just delay the creation of popart:inferenceSession to the first odla_CreateContext()